Speaker screen cubit

TreavVasu opened the pull request:
## Fixes #46 

### Description
Implemented SpeakerCubit 
Added FakeSpeakerRepository
Tested both

**Type of Change:** (Delete irrelevant options)
- Code
- Testing

**Code/Quality Assurance Only** (Delete irrelevant options)

- New feature (non-breaking change which adds functionality pre-approved by mentors)


### How Has This Been Tested?


![untitled](https://user-images.githubusercontent.com/16835617/105809186-3d861b00-5fcf-11eb-8059-cb0e811e6b10.gif)


### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] I have added one/more required dependencies
- [x] I have formatted the code using dart format (dartfmt)


**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

TreavVasu committed:
Package Config

TreavVasu committed:
package config

TreavVasu committed:
API and Sharedpref implementation API Untested

TreavVasu committed:
Merge branch 'main' of https://github.com/ECellNitrr/ECellApp2021 into Login-API

TreavVasu committed:
updating with upstream

TreavVasu committed:
feat:LoginApi

TreavVasu committed:
Update lib/main.dart

Co-authored-by: Viren Manojkumar Khatri <werainkhatri.work@gmail.com>

TreavVasu committed:
Update lib/main.dart

Co-authored-by: Viren Manojkumar Khatri <werainkhatri.work@gmail.com>

TreavVasu committed:
Update lib/screens/login/login_repository.dart

Co-authored-by: Viren Manojkumar Khatri <werainkhatri.work@gmail.com>

TreavVasu committed:
Update:login_repository

TreavVasu committed:
feat: Login API

TreavVasu committed:
Merge branch 'main' of https://github.com/ECellNitrr/ECellApp2021 into Login-API

TreavVasu committed:
Minor fix

TreavVasu committed:
Added  pubspec.lock  to .gitignore

TreavVasu committed:
Merge branch 'main' of https://github.com/ECellNitrr/ECellApp2021 into Login-API

TreavVasu committed:
Resolving issues

TreavVasu committed:
Minor fix

TreavVasu committed:
Merge branch 'main' of https://github.com/ECellNitrr/ECellApp2021 into Login-API

TreavVasu committed:
feat:Speaker Cubit

TreavVasu committed:
feat:SpeakerCubit

TreavVasu committed:
pubspec.lock issue fix

TreavVasu committed:
feat:Speaker Cubit

werainkhatri commented on lib/models/speaker.dart:
```diff
@@ -0,0 +1,56 @@
+import 'package:equatable/equatable.dart';
+
+class Speaker extends Equatable {
+  final String comapny;
```
spelling mistake
```suggestion
  final String company;
```

werainkhatri replied:
🔴 
This is still incorrectly changed. Please be careful while making changes.

werainkhatri commented on lib/models/speaker.dart:
```diff
@@ -0,0 +1,56 @@
+class Speaker extends Equatable {
+  final String comapny;
+  final String contact;
+  final String createdAt;
```
don't add fields that are not relevant for the app. Discuss with the backend team / refer the designs and add only those fields that are needed to be displayed in the app. Also remove them from `strings.dart`

werainkhatri commented on lib/models/speaker.dart:
```diff
@@ -0,0 +1,56 @@
+
+  @override
+  // TODO: implement props
+  List<Object> get props => [];
```
id should be the prop

werainkhatri commented on lib/screens/speaker/speaker_repository.dart:
```diff
@@ -0,0 +1,46 @@
+
+class FakeSpeakerRepository implements SpeakerRepository {
+  @override
+  Future<List<dynamic>> speakers() async {
```
give better names to functions.
```suggestion
  Future<List<dynamic>> getSpeakerList() async {
```

werainkhatri commented on lib/screens/speaker/speaker_repository.dart:
```diff
@@ -0,0 +1,46 @@
+@immutable
+abstract class SpeakerRepository {
+  /// All subfunctions are final No arguments required returns json
+  Future<List<dynamic>> speakers();
```
use the `fromJson()` function created in the `Speaker` class and return list of speakers
```suggestion
  Future<List<Speaker>> speakers();
```

werainkhatri commented on pubspec.lock:
```diff
@@ -7,7 +7,7 @@ packages:
       name: async
```
🔴 
commit the pubspec.lock changes only if you have changed the pubspec.yaml file

werainkhatri commented on .gitignore:
```diff
@@ -30,8 +30,7 @@
 /build/
-.lock 
-pubspec.lock
+*.lock
```
remove it from here.

werainkhatri commented on lib/models/speaker.dart:
```diff
@@ -0,0 +1,56 @@
+  final int id;
+  final String modifiedAt;
+  final String name;
+  final String profilePic;
```
add doc comments on non-obvious variables as to what kind of value they hold

TreavVasu replied:
From the result of a conversation with the backend team, I will be removing unnecessary fields that also include but not limited to email and contact.

werainkhatri commented on lib/screens/speaker/cubit/speaker_state.dart:
```diff
@@ -0,0 +1,26 @@
+}
+
+class SpeakerSuccess extends SpeakerState {
+  final List<dynamic> json;
```
should hold list of speakers

werainkhatri commented on lib/screens/speaker/speaker.dart:
```diff
@@ -0,0 +1,80 @@
+          speakerResponse = state.json;
+          return _buildSuccess(context);
+        } else if (state is SpeakerLoading) {
+          _speaker();
```
since builder is called multiple times depending on how the ui is needed to be built and how flutter builds ui, non-ui functions should be called outside it, either in listener or initState. in this case, since there is a case where the _getSpeakers (change the name) function is to be called again (retry in case of error) it should be called in listener.

werainkhatri commented on lib/screens/speaker/speaker.dart:
```diff
@@ -0,0 +1,80 @@
+  Widget _buildAskReload() {
+    //Ask to reload screen
+    //TODO: Implement a Screen to reload
+    return Container();
```
here, there should be a retry button that calls _getSpeaker function. coordinate with the designer on how this should look.

werainkhatri commented on lib/screens/speaker/speaker.dart:
```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class SpeakerScreen extends StatefulWidget {
```
since there are no setState function calls required, this should be a stateless widget

TreavVasu replied:
I applied StatefulWrapper is it still required now?

werainkhatri replied:
The reason for using stateful wrapper was so that we can have an init function in a stateless widget. You still have to make it stateless.

TreavVasu committed:
Minor fix

TreavVasu committed:
Merge branch 'main' of https://github.com/ECellNitrr/ECellApp2021 into SpeakerScreen-Cubit

TreavVasu committed:
Resolving patch

TreavVasu committed:
Fix

TreavVasu committed:
Merge branch 'main' of https://github.com/ECellNitrr/ECellApp2021 into SpeakerScreen-Cubit

TreavVasu committed:
.

TreavVasu committed:
.

TreavVasu committed:
Added StatefulWrapper

TreavVasu committed:
.

TreavVasu committed:
.

TreavVasu committed:
..

TreavVasu committed:
...

TreavVasu committed:
Fixed

TreavVasu committed:
fix

TreavVasu committed:
fix

werainkhatri commented:
Go over all the unresolved conversations from above and fix them.

TreavVasu committed:
Issue74 Minor fix

TreavVasu committed:
Issue74 Minor fix

TreavVasu committed:
Fix:Minor changes

Changes in doc to help better understand .
Remove Unnecessary api keys
Corrected minor errors in naming

TreavVasu committed:
Merge branch 'SpeakerScreen-Cubit' of https://github.com/TreavVasu/ECellApp2021 into SpeakerScreen-Cubit

TreavVasu committed:
Organizing files to commit
fixing change

TreavVasu committed:
Re-attached  Login UI with Start
De attached SpeakerScreen with start

TreavVasu committed:
Re-attached  Login UI with Start
De attached SpeakerScreen from start

TreavVasu committed:
Minor fix
Removed Unused imports

TreavVasu committed:
Replaced stateful with stateless
Final Commit

TreavVasu committed:
Final commit

TreavVasu committed:
Final fix
Discarded unused code
